kernel/src/process_load.rs

```diff
@@ -0,0 +1,791 @@
+            }
+
+            ProcessLoadError::MemoryAddressMismatch {
+                actual_address,
+                expected_address,
+            } => write!(
+                f,
+                "App memory does not match requested address Actual:{:#x}, Expected:{:#x}",
+                actual_address, expected_address
+            ),
+
+            ProcessLoadError::IncorrectFlashAddress {
+                actual_address,
+                expected_address,
+            } => write!(
+                f,
+                "App flash does not match requested address. Actual:{:#x}, Expected:{:#x}",
+                actual_address, expected_address
+            ),
+
+            ProcessLoadError::IncompatibleKernelVersion { version } => match version {
+                Some((major, minor)) => write!(
+                    f,
+                    "Process is incompatible with the kernel. Running: {}.{}, Requested: {}.{}",
+                    crate::MAJOR,
+                    crate::MINOR,
+                    major,
+                    minor
+                ),
+                None => write!(f, "Process did not provide a TBF kernel version header"),
+            },
+
+            ProcessLoadError::InternalError => write!(f, "Error in kernel. Likely a bug."),
+
+            ProcessLoadError::CredentialsNoAccept => write!(f, "No credentials accepted."),
+
+            ProcessLoadError::CredentialsReject(index) => {
+                write!(f, "Credentials index {} rejected.", index)
+            }
+        }
+    }
+}
+
+/// Load processes (stored as TBF objects in flash) into runnable
+/// process structures stored in the `procs` array. If a `checker` is
+/// passed, this method scans the footers in the TBF for cryptographic
+/// credentials for binary integrity, passing them to `checker` to decide
+/// whether the process has sufficient credentials to run.
+#[inline(always)]
+#[allow(dead_code)]
+pub fn load_and_check_processes<C: Chip>(
+    kernel: &'static Kernel,
+    chip: &'static C,
+    app_flash: &'static [u8],
+    app_memory: &'static mut [u8],
+    mut procs: &'static mut [Option<&'static dyn Process>],
+    fault_policy: &'static dyn ProcessFaultPolicy,
+    capability_management: &dyn ProcessManagementCapability,
+) -> Result<(), ProcessLoadError> {
+    load_processes(
+        kernel,
+        chip,
+        app_flash,
+        app_memory,
+        &mut procs,
+        fault_policy,
+        capability_management,
+    )?;
+    let _res = check_processes(procs, kernel);
+    Ok(())
+}
+
+/// Load processes (stored as TBF objects in flash) into runnable
+/// process structures stored in the `procs` array. If a `checker` is
+/// passed, this method scans the footers in the TBF for cryptographic
+/// credentials for binary integrity, passing them to `checker` to decide
+/// whether the process has sufficient credentials to run.
+#[inline(always)]
+#[allow(dead_code)]
```

Comment:
here too

Reply:
Same. A board will call one of these two (`load_processes`, `load_and_check_processes`) methods.

Reply:
But it's public from this crate. I removed both allow dead code and didn't see any warnings.